### PR TITLE
Deploy flags Documentation

### LINKED ISSUE #278: Additional apsody operator and appsody deploy documentation for deploy flags and coexistence
The following need to be documented:

Coexistence of appsody operators created by the CLI with appsody operators created by other means.

The use of appsody deploy --generate-only with the --knative and --tag(-t) flags.

### PR DESCRIPTION
**Pull request by kewegner**
Fixes #278

### CODE REVIEW

File: content/docs/using-appsody/building-and-deploying.md [hunk 1]
```diff
@@ -179,6 +186,13 @@ rules:
 
   Once you have these roles in place, you need to create the appropriate RoleBinding and ClusterRoleBinding to bind your users or groups to them.
 
+
+#### Coexistence of an appsody operator created by appsody operator install or appsody deploy with appsody operators created by other means
```
Comment (chilanti) [inline content/docs/using-appsody/building-and-deploying.md]: I'd rephrase this heading... "Considerations on Appsody operators not installed via the Appsody CLI"

File: content/docs/using-appsody/building-and-deploying.md [hunk 2]
```diff
@@ -179,6 +186,13 @@ rules:
...
+
+- Only one appsody operator per namespace is allowed when created by `appsody operator install` or `appsody deploy`.  The code for `appsody operator install` and `appsody deploy` ensures this is the case.  If an appsody operator is created by another means, it is up to the user to make sure that only one appsody operator exists in the given namespace.
```
Comment (chilanti) [inline content/docs/using-appsody/building-and-deploying.md]: And add an intro line: "Appsody operators can be installed through different means - the Appsody CLI is one of the ways to get them installed."

Comment (chilanti) [reply content/docs/using-appsody/building-and-deploying.md]: I would simplify: "The Appsody CLI assumes you have at most one Appsody operator per namespace. If you install the operator with the Appsody CLI, this constraint is enforced by the CLI itself. If you use different techniques to install the operator, make sure you do not install multiple operators in the same namespace."

File: content/docs/using-appsody/building-and-deploying.md [hunk 3]
```diff
@@ -179,6 +187,13 @@ rules:
 
   Once you have these roles in place, you need to create the appropriate RoleBinding and ClusterRoleBinding to bind your users or groups to them.
 
+
+#### Considerations on Appsody operators not installed via the Appsody CLI
+Appsody operators can be installed through different means - the Appsody CLI is one of the ways to get them installed.
+- The Appsody CLI assumes you have at most one Appsody operator per namespace. If you install the operator with the Appsody CLI, this constraint is enforced by the CLI itself. If you use different techniques to install the operator, make sure you do not install multiple operators in the same namespace.
+
+- The appsody operators created by `appsody operator install` or `appsody deploy` watch only one namespace.  However, the appsody CLI operations `appsody operator install` and `appsody deploy` can tolerate an appsody operator created by a different mechanism which watches multiple namespaces.
```
Comment (helenmasters) [inline content/docs/using-appsody/building-and-deploying.md]: appsody operators -> Appsody operators
appsody CLI -> Appsody CLI
appsody operator -> Appsody operator

File: content/docs/using-appsody/building-and-deploying.md [hunk 4]
```diff
@@ -97,6 +97,14 @@ appsody deploy delete
 ```
 This command must be run from your Appsody project directory. It will attempt to remove the application you installed, but it will not touch the Appsody operator.
 
+Here we discuss the --generate-only flag, which allows you to create the deployment manifest without triggering a deployment.
+#### Use of  `appsody deploy --generate-only` flag with the `--tag  (-t)` flag 
+If the `--generate only` flag is used in conjunction with the `--tag (-t)` flag, the value of the tag is preserved in the app-deploy.yaml file and used during subsequent `appsody deploy` invocations.
+
+#### Use of `appsody deploy --generate-only` with the `--knative` tag 
+If --knative is used along with --generate-only, the value of `createKnativeService` will be set to `true` in the generated app-deploy.yaml file.
```
Comment (helenmasters) [inline content/docs/using-appsody/building-and-deploying.md]: We should be consistent here, therefore the flags such as --knative and --generate-only should be backtick'd in all the sections here.

File: content/docs/using-appsody/building-and-deploying.md [hunk 5]
```diff
@@ -97,6 +97,14 @@ appsody deploy delete
 ```
 This command must be run from your Appsody project directory. It will attempt to remove the application you installed, but it will not touch the Appsody operator.
 
+Here we discuss the --generate-only flag, which allows you to create the deployment manifest without triggering a deployment.
+#### Use of  `appsody deploy --generate-only` flag with the `--tag  (-t)` flag 
+If the `--generate only` flag is used in conjunction with the `--tag (-t)` flag, the value of the tag is preserved in the app-deploy.yaml file and used during subsequent `appsody deploy` invocations.
+
+#### Use of `appsody deploy --generate-only` with the `--knative` tag 
+If --knative is used along with --generate-only, the value of `createKnativeService` will be set to `true` in the generated app-deploy.yaml file.  
+Note that if during a subsequent appsody deploy call the --knative flag is not present, the value of `createKnativeService` will be set to `false` in app-deploy.yaml.
```
Comment (helenmasters) [inline content/docs/using-appsody/building-and-deploying.md]: this reference should either be Appsody deploy or `appsody deploy` if you're referring to the command.

**Comment by kewegner** [comment]
@helenmasters can you review this again?
